cmd/otel-allocator/internal/collector/collector.go

```diff
@@ -23,33 +23,32 @@ const (
 	defaultMinUpdateInterval = time.Second * 5
 )
 
-var (
-	collectorsDiscovered = promauto.NewGauge(prometheus.GaugeOpts{
-		Name: "opentelemetry_allocator_collectors_discovered",
-		Help: "Number of collectors discovered.",
-	})
-)
-
 type Watcher struct {
 	log                          logr.Logger
 	k8sClient                    kubernetes.Interface
 	close                        chan struct{}
 	minUpdateInterval            time.Duration
 	collectorNotReadyGracePeriod time.Duration
+	collectorsDiscovered         metric.Int64Gauge
 }
 
-func NewCollectorWatcher(logger logr.Logger, kubeConfig *rest.Config, collectorNotReadyGracePeriod time.Duration) (*Watcher, error) {
+func NewCollectorWatcher(logger logr.Logger, meter metric.Meter, kubeConfig *rest.Config, collectorNotReadyGracePeriod time.Duration) (*Watcher, error) {
 	clientset, err := kubernetes.NewForConfig(kubeConfig)
 	if err != nil {
 		return &Watcher{}, err
 	}
 
+	collectorsDiscovered, err := meter.Int64Gauge("opentelemetry_allocator_collectors_discovered", metric.WithDescription("Number of collectors discovered."))
```

Comment:
i forget the recommendation, but should we just use the global meter here (and the other constructors)?

Reply:
well, we create the meter provider in main, we pass it down in the objects - maybe we are creating the gauge multiple times? I don't know.

Reply:
there's a global provider?

Reply:
ok, will use that instead!